src_assets/common/assets/web/index.html

```diff
@@ -89,16 +89,15 @@ <h2>Legal</h2>
       try {
         this.version = (await fetch("/api/config").then((r) => r.json())).version;
         this.githubVersion = (await fetch("https://api.github.com/repos/LizardByte/Sunshine/releases/latest").then((r) => r.json()));
-        if (this.version.split("-g").length > 1) {
+        if (!this.buildVersionIsStable()) {
```

Comment:
This is not enough to determine if it's a nightly version... and in fact will always return false since checking for stable version checks if `this.nightlyData` is populated.

![image](https://user-images.githubusercontent.com/42013603/227789502-4b7690fa-63e8-4a2a-8d6c-be4d6ff97267.png)


But this will work.

Change L92 to `if (this.buildVersionIsNightly()) {`

And add a new method

```js
methods: {
      buildVersionIsStable() {
        return this.version.split(".").length === 3 || !this.version || !this.nightlyData;
      },
      buildVersionIsNightly() {
        return this.version.split("."). length === 4 || !this.version
      }
    }
```

These two changes result in the code working correctly.

![image](https://user-images.githubusercontent.com/42013603/227789607-5a7396cf-faae-4f36-a1b2-98064c42651e.png)

Reply:
in JavaScript, booleans will short circuit. So it shouldn’t even test !nightlyData because the first check is checking the version split.

I’m not sure why you’re running into errors again though

Reply:
Never mind it’s not short circuiting in this context, I’ll fix it later in a few hours.

Reply:
My first thought here is there is no need to even check nightly data to determine if it’s nightly/stable

Reply:
Maybe not.

If it helps this is the logic that determines what version is used during compilation.

https://github.com/LizardByte/Sunshine/blob/a487fb31eaaa47c98bfab45eddaf34e33f2932ab/CMakeLists.txt#L14-L71

Reply:
They're still needed, sort of. The main reason is that UI might try to load before the data is fully loaded. It's mostly because of how the fetch requests are being done. I will refactor those at a later time but I fixed the issues now.